air/src/runner.rs

```diff
@@ -64,7 +64,13 @@ fn execute_air_impl(
 
     // match here is used instead of map_err, because the compiler can't determine that
     // they are exclusive and would treat exec_ctx and trace_handler as moved
-    match air.execute(&mut exec_ctx, &mut trace_handler) {
+    match {
+        // create a span for tracing the whole air.execute and its handling
+        let execute_span = tracing::span!(tracing::Level::INFO, "execute");
+        let _enter = execute_span.enter();
+
+        air.execute(&mut exec_ctx, &mut trace_handler)
+    } {
```

Comment:
Let's add a new variable and then match on it.

Reply:
It will be solved by the macro.